1_fetch/src/get_wqp_inventory.R

```diff
@@ -27,23 +28,128 @@ inventory_wqp <- function(grid, char_names, ...){
-    dataRetrieval::whatWQPdata(
-      bBox = c(bbox$xmin, bbox$ymin, bbox$xmax, bbox$ymax),
-      characteristicName = x,
-      ...) %>%
-      mutate(CharacteristicName = x)
+    # define arguments for whatWQPdata
+    wqp_args_all <- c(wqp_args, list(bBox = c(bbox$xmin, bbox$ymin, bbox$xmax, bbox$ymax),
+                                     characteristicName = x))
+    # query WQP
+    dataRetrieval::whatWQPdata(wqp_args_all) %>%
+      mutate(CharacteristicName = x, grid_id = grid$id)
   }) %>%
     bind_rows()
   
-  # Format WQP inventory 
+  # Fetch missing CRS information from WQP
+  site_location_metadata <- dataRetrieval::whatWQPsites(
+    bBox = c(bbox$xmin, bbox$ymin, bbox$xmax, bbox$ymax)) %>%
+    select(MonitoringLocationIdentifier, HorizontalCoordinateReferenceSystemDatumName) %>%
+    filter(MonitoringLocationIdentifier %in% wqp_inventory$MonitoringLocationIdentifier)
+  
+  # Join WQP inventory with site metadata 
   wqp_inventory_out <- wqp_inventory %>%
-    group_by(CharacteristicName) %>%
-    summarize(sites_count = n(),
-              results_count = sum(resultCount)) %>%
-    mutate(grid_id = grid$id) %>%
-    select(grid_id, CharacteristicName, sites_count, results_count)
+    left_join(site_location_metadata, by = "MonitoringLocationIdentifier")
 
   return(wqp_inventory_out)
   
 }
 
+
+
+#' Function to transform WQP sites to use a consistent coordinate reference system
+#'
+#' @param sites data frame containing sites returned from the WQP query; contains
+#' columns lon, lat, and HorizontalCoordinateReferenceSystemDatumName.
+#' @param crs_out character string indicating the desired output coordinate reference
+#' system (crs); options include "NAD83" or "WGS84", defaults to "WGS84"
+#'
+#' @examples
+#' transform_site_locations(sites)
+#' transform_site_locations(sites, "NAD83")
+#' 
+transform_site_locations <- function(sites, crs_out = "WGS84"){
+  
+  message("Attempting to harmonize different site CRS...")
+  
+  # Define EPSG code for output data frame 
+  epsg_out <- case_when(crs_out == "NAD83" ~ 4269,
+                        crs_out == "WGS84" ~ 4326)
+  
+  # Transform sites into a consistent crs defined by epsg_out
+  sites_transformed <- sites %>%
+    # transform the sites data frame into a list where all sites within each 
+    # list element share the same crs/datum
+    rename(datum = HorizontalCoordinateReferenceSystemDatumName) %>%
+    split(.,.$datum) %>%
+    # Each list element contains sites with different crs. Transform the sites 
+    # within each element to the desired crs
+    lapply(., function(x){
+      # assume unknown crs (datum equals "UNKNWN", "OTHER") correspond with WGS84
+      epsg_in <- case_when(x$datum[1] == "NAD83" ~ 4269,
+                           x$datum[1] == "WGS84" ~ 4326,
+                           x$datum[1] == "NAD27" ~ 4267,
+                           x$datum[1] == "UNKWN" ~ 4326,
+                           x$datum[1] == "OTHER" ~ 4326)
```

Comment:
When I expect a column to have all the same values, I usually use `unique()` rather than `[1]` just in case somewhere along the way, we end up breaking that assumption and not getting any error or have any knowledge about it.

```suggestion
      epsg_in <- case_when(unique(x$datum) == "NAD83" ~ 4269,
                           unique(x$datum) == "WGS84" ~ 4326,
                           unique(x$datum) == "NAD27" ~ 4267,
                           unique(x$datum) == "UNKWN" ~ 4326,
                           unique(x$datum) == "OTHER" ~ 4326)
```

Reply:
Yes, this makes sense! I've taken your suggestion to swap `[1]` with `unique()` here.